src/CMakeLists.txt

```diff
@@ -33,8 +35,25 @@ if(CMAKE_EXTRA_GENERATOR OR MSVC_IDE)
     list(APPEND CUKE_SOURCES ${CUKE_HEADERS})
 endif()
 
-add_library(cucumber-cpp-nomain STATIC ${CUKE_SOURCES})
-add_library(cucumber-cpp STATIC ${CUKE_SOURCES} main.cpp)
+if(CUKE_ENABLE_SHARED_LIB)
+    set(CUKE_LINKAGE SHARED)
+else()
+    set(CUKE_LINKAGE STATIC)
+endif()
+
+add_library(cucumber-cpp-nomain ${CUKE_LINKAGE} ${CUKE_SOURCES})
+add_library(cucumber-cpp ${CUKE_LINKAGE} ${CUKE_SOURCES} main.cpp)
+
+set_target_properties(cucumber-cpp cucumber-cpp-nomain PROPERTIES
+    DEFINE_SYMBOL cucumber_cpp_EXPORTS
+)
+
+generate_export_header(cucumber-cpp
+    EXPORT_FILE_NAME "CukeExport.hpp"
```

Comment:
In order to place this in the same, relative, include directory structure, I think we should use `EXPORT_FILE_NAME cucumber-cpp/internal/CukeExport.h` (this will affect the required include directives which will need to change to `#include <cucumber-cpp/internal/CukeExport.h>`). This would make library installs place the headers all in a single location.